--- NDK_3.9/Include/include_h/clib/alib_stdio_protos.h	2001-12-17 15:14:44.000000000 +0100
+++ NDK_3.9/Include/include_h/clib/alib_stdio_protos.h	2018-01-22 19:33:55.310761400 +0100
@@ -21,16 +21,16 @@
 
 /* stdio functions that duplicate those in a typical C library */
 
-LONG printf( CONST_STRPTR fmt, ... );
-LONG sprintf( CONST_STRPTR buffer, CONST_STRPTR fmt, ... );
-LONG fclose( LONG stream );
-LONG fgetc( LONG stream );
-LONG fprintf( LONG stream, CONST_STRPTR fmt, ... );
-LONG fputc( LONG c, LONG stream );
-LONG fputs( CONST_STRPTR s, LONG stream );
-LONG getchar( VOID );
-LONG putchar( ULONG c );
-LONG puts( CONST_STRPTR s );
+__stdargs LONG printf( CONST_STRPTR fmt, ... );
+__stdargs LONG sprintf( CONST_STRPTR buffer, CONST_STRPTR fmt, ... );
+__stdargs LONG fclose( LONG stream );
+__stdargs LONG fgetc( LONG stream );
+__stdargs LONG fprintf( LONG stream, CONST_STRPTR fmt, ... );
+__stdargs LONG fputc( LONG c, LONG stream );
+__stdargs LONG fputs( CONST_STRPTR s, LONG stream );
+__stdargs LONG getchar( VOID );
+__stdargs LONG putchar( ULONG c );
+__stdargs LONG puts( CONST_STRPTR s );
 
 #ifdef __cplusplus
 }
