--- cdplayer.library/include/fd/cdplayer.fd.orig	2022-04-22 22:53:12.738869052 -0700
+++ cdplayer.library/include/fd/cdplayer.fd	2022-04-22 22:53:38.970284940 -0700
@@ -1,15 +1,15 @@
 ##base _CDPlayerBase
 ##bias 30
-CDEject(io_ptr) (a5)
-CDPlay(starttrack,endtrack,io_ptr) (a0,a1,a5)
-CDResume(Mode,io_ptr) (a0,a5)
-CDStop(io_ptr) (a5)
-CDJump(Blocks,io_ptr) (a0,a5)
-CDActive(io_ptr) (a5)
-CDCurrentTitle(io_ptr) (a5)
-CDTitleTime(cd_time,io_ptr) (a0,a5)
-CDGetVolume(vol,io_ptr) (a0,a5)
-CDSetVolume(vol,io_ptr) (a0,a5)
-CDReadTOC(toc,io_ptr) (a0,a5)
-CDInfo(cdi,io_ptr) (a0,a5)
+CDEject(io_ptr)(a5)
+CDPlay(starttrack,endtrack,io_ptr)(a0,a1,a5)
+CDResume(Mode,io_ptr)(a0,a5)
+CDStop(io_ptr)(a5)
+CDJump(Blocks,io_ptr)(a0,a5)
+CDActive(io_ptr)(a5)
+CDCurrentTitle(io_ptr)(a5)
+CDTitleTime(cd_time,io_ptr)(a0,a5)
+CDGetVolume(vol,io_ptr)(a0,a5)
+CDSetVolume(vol,io_ptr)(a0,a5)
+CDReadTOC(toc,io_ptr)(a0,a5)
+CDInfo(cdi,io_ptr)(a0,a5)
 ##end
