--- gfxbase.h	2001-12-17 14:13:22.000000000 +0100
+++ gfxbase0.h	2018-10-24 16:48:00.915542000 +0200
@@ -65,6 +65,9 @@
 	UWORD	*LCMptr;
 	UWORD	MicrosPerLine;		/* 256 time usec/line */
 	UWORD	MinDisplayColumn;
+#ifdef __KICK13__
+	ULONG	reserved[23];	 /* for future use */
+#else
 	UBYTE	ChipRevBits0;
 	UBYTE	MemType;
 	UBYTE	crb_reserved[4];
@@ -125,6 +128,7 @@
 	ULONG	VBCounter;
 	struct	SignalSemaphore *HashTableSemaphore;
 	ULONG	*HWEmul[9];
+#endif
 };
 
 #define ChunkyToPlanarPtr HWEmul[0]
