--- old/devices/timer.h	2001-12-17 14:03:44.000000000 +0100
+++ new/devices/timer.h	2020-04-13 20:28:34.518085400 +0200
@@ -27,10 +27,19 @@
 
 #define TIMERNAME	"timer.device"
 
+#ifndef _TIMEVAL_DEFINED
+#define _TIMEVAL_DEFINED
 struct timeval {
-    ULONG tv_secs;
-    ULONG tv_micro;
+    union {
+        ULONG          tv_sec;         /* seconds */
+        ULONG          tv_secs;
+    };
+    union {
+        ULONG     tv_usec;        /* and microseconds */
+        ULONG     tv_micro;
+    };
 };
+#endif
 
 struct EClockVal {
     ULONG ev_hi;
